File: backend/src/models/tournament.ts
import { sequelize } from '../database';
import { DataTypes, ModelDefined, Optional } from 'sequelize';

interface Tournament {
  id: string;
  name: string;
  startDate: Date;
  endDate: Date;
  isActive: boolean;
  tournamentLogo: string;
  status: 'upcoming' | 'ongoing' | 'completed';
}

type TournamentCreationAttributes = Optional<Tournament, 'id'>;

const TournamentModel: ModelDefined<Tournament, TournamentCreationAttributes> = sequelize.define('tournaments', {
  id: {
    type: DataTypes.UUID,
    primaryKey: true,
    defaultValue: DataTypes.UUIDV4, // Automatically generate a UUID
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  startDate: {
    type: DataTypes.DATE,
    allowNull: false,
  },
  endDate: {
    type: DataTypes.DATE,
    allowNull: false,
  },
  tournamentLogo:{
    type: DataTypes.STRING,
    allowNull: true
  },
  status: {
    type: DataTypes.ENUM('upcoming', 'ongoing', 'completed'),
    allowNull: false,
  },
  isActive: {
    type: DataTypes.BOOLEAN,
    defaultValue: true
  }
});


export { TournamentModel };

File: yorker/lib/models/tournament.dart
class Tournament {
  final String id;
  final String name;
  final String startDate;
  final String endDate;
  final String tournamentLogo;
  final String status;
  final bool isActive;
  final String createdAt;
  final String updatedAt;

  Tournament({
    required this.id,
    required this.name,
    required this.startDate,
    required this.endDate,
    required this.tournamentLogo,
    required this.status,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Tournament.fromJson(Map<String, dynamic> json) {
    return Tournament(
      id: json['id'],
      name: json['name'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      tournamentLogo: json['tournamentLogo'],
      status: json['status'],
      isActive: json['isActive'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
    );
  }
}
